src/utils/PolygonConnector.cpp

```diff
@@ -0,0 +1,271 @@
+/** Copyright (C) 2018 Ultimaker - Released under terms of the AGPLv3 License */
+
+#include "PolygonConnector.h"
+
+#include "linearAlg2D.h"
+
+namespace cura 
+{
+
+Polygons PolygonConnector::connect()
+{
+    Polygons ret;
+    
+    std::vector<Polygon> to_connect;
+    to_connect.reserve(input_polygons.size());
+    for (ConstPolygonPointer poly : input_polygons)
+    {
+        to_connect.emplace_back(*poly); // copy into list
+    }
+
+    while (!to_connect.empty())
+    {
+        Polygon current = std::move(to_connect.back());
+        to_connect.pop_back();
+
+        std::optional<PolygonBridge> bridge = getBridge(current, to_connect);
+        if (bridge)
+        {
+            all_bridges.push_back(*bridge); // just for keeping scores
```

Comment:
Hmm I don't know.

The tests depend on this. There are tests in the test suite which test the created bridges, but if the bridges are not separately stored there is no way to deduce from a connected polygon where the bridges were.

I don't think we should refactor the whole code so that the test case can test the code without the need of a `all_bridges`, because that would require quite a refactor while it is just for testing purposes.

One option would be to construct the class with some variable `debug_mode = true` and only then add the bridge to the list.

What do you think?

Reply:
I think we can add then "// just for keeping scores. This is needed for test cases", 
Then it is easier to understand the reason.

Reply:
But then the test cases won't compile!
You'll get compile errors.

I always compile the test cases... I don't want to have to change code every time I compile test cases.

Reply:
I mean, let's keep the tack all_bridges
Only change:
all_bridges.push_back(*bridge); // just for keeping scores. This is needed for test cases.

Reply:
Oh wait the test case doesn't fail, but half of the tests become superfluous because they don't really test what they should be testing any more.